### ISSUE TITLE
Bad subgraphs in snapshot hca_dev_20200819___20201105_optimusb

### ISSUE THREAD

**Issue by hannes-ucsc** [DataBiosphere/azul#2430]
Subgraphs with the same error error from reindex of dev from #2434:
- c41dffbf-ad83-447c-a0e1-13e689d9b258
- ad98d3cd-26fb-4ee3-99c9-8a2ab085e737
- 83f5188e-3bf7-4956-9544-cea4f8997756
- 7027adc6-c9c9-46f3-84ee-9badc3a4f53b
- 14c7a4c3-b01f-5923-afa5-ad2ce304ee6d
- 2086eb05-10b9-432b-b7f0-169ccc49d270
- 455b46e6-d8ea-4611-861e-de720a562ada
- 95f07e6e-6a73-4e1b-a880-c83996b3aa5c
- a815c84b-8999-433f-958e-422c0720e00d
- b4a7d12f-6c2f-40a3-9e35-9756997857e3
- c1a9a93d-d9de-4e65-9619-a9cec1052eaa
- 2ef3655a-973d-4d69-9b41-21fa4041eed7
- f2fe82f0-4454-4d84-b416-a885f3121e59
- 38449aea-70b5-40db-84b3-1e08f32efe34
- 42d4f8d4-5422-4b78-adae-e7c3c2ef511c
- f48e7c39-cc67-4055-9d79-bc437892840c

----

Subgraph `42d4f8d4-5422-4b78-adae-e7c3c2ef511c` cannot be indexed:

**CloudWatch Logs Insights**  
region: us-east-1  
log-group-names: /aws/lambda/azul-indexer-sandbox-contribute  
start-time: -3600s  
end-time: 0s  
query-string:
```
fields @timestamp, @message
| filter @message like "ef4175e9-fc8e-50f0-834a-2df7057875a7"
| sort @timestamp desc
| limit 20
```
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
|       @timestamp        |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 @message                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
|-------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 2020-11-06 07:14:05.790 | END RequestId: ef4175e9-fc8e-50f0-834a-2df7057875a7                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
| 2020-11-06 07:14:05.790 | REPORT RequestId: ef4175e9-fc8e-50f0-834a-2df7057875a7 Duration: 7602.63 ms Billed Duration: 7700 ms Memory Size: 256 MB Max Memory Used: 176 MB                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
| 2020-11-06 07:14:05.783 | [INFO] 2020-11-06T07:14:05.782Z ef4175e9-fc8e-50f0-834a-2df7057875a7 Retrieved 1 entities of type 'supplementary_file'                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
| 2020-11-06 07:14:05.783 | [WARNING] 2020-11-06T07:14:05.783Z ef4175e9-fc8e-50f0-834a-2df7057875a7 Worker failed to handle message {'action': 'add', 'notification': {'query': {}, 'subscription_id': 'cafebabe-feed-4bad-dead-beaf8badf00d', 'transaction_id': 'bc171f20-b71c-4f31-8fc9-de41e88424ab', 'match': {'bundle_uuid': '42d4f8d4-5422-4b78-adae-e7c3c2ef511c', 'bundle_version': '2020-10-07T11:11:17.095956Z'}}, 'catalog': 'dcp2'}. Traceback (most recent call last):   File "/var/task/azul/indexer/index_controller.py", line 145, in contribute     contributions = self.transform(catalog, notification, delete)   File "/var/task/azul/indexer/index_controller.py", line 172, in transform     bundle = self.repository_plugin(catalog).fetch_bundle(bundle_fqid)   File "/var/task/azul/plugins/repository/tdr/__init__.py", line 122, in fetch_bundle     bundle = self.emulate_bundle(bundle_fqid)   File "/var/task/azul/plugins/repository/tdr/__init__.py", line 217, in emulate_bundle     raise e   File "/var/lang/lib/python3.8/concurrent/futures/thread.py", line 57, in run     result = self.fn(*self.args, **self.kwargs)   File "/var/task/azul/plugins/repository/tdr/__init__.py", line 298, in _retrieve_entities     require(not missing,   File "/var/task/azul/__init__.py", line 1050, in require     reject(not condition, *args, exception=exception)   File "/var/task/azul/__init__.py", line 1065, in reject     raise exception(*args) azul.RequirementError: Required entities not found in hca_dev_20200819___20201105_optimusb.project: {'42d4f8d4-5422-4b78-adae-e7c3c2ef511c'} |
| 2020-11-06 07:14:03.071 | [ERROR] 2020-11-06T07:14:03.56Z ef4175e9-fc8e-50f0-834a-2df7057875a7 TDR worker failed to retrieve entities of type 'project' Traceback (most recent call last):   File "/var/lang/lib/python3.8/concurrent/futures/thread.py", line 57, in run     result = self.fn(*self.args, **self.kwargs)   File "/var/task/azul/plugins/repository/tdr/__init__.py", line 298, in _retrieve_entities     require(not missing,   File "/var/task/azul/__init__.py", line 1050, in require     reject(not condition, *args, exception=exception)   File "/var/task/azul/__init__.py", line 1065, in reject     raise exception(*args) azul.RequirementError: Required entities not found in hca_dev_20200819___20201105_optimusb.project: {'42d4f8d4-5422-4b78-adae-e7c3c2ef511c'}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |
| 2020-11-06 07:14:03.056 | [INFO] 2020-11-06T07:14:03.55Z ef4175e9-fc8e-50f0-834a-2df7057875a7 Retrieved 0 entities of type 'project'                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
| 2020-11-06 07:14:03.056 | [INFO] 2020-11-06T07:14:03.56Z ef4175e9-fc8e-50f0-834a-2df7057875a7 Retrieving 1 entities of type 'supplementary_file' ...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
| 2020-11-06 07:14:00.555 | [INFO] 2020-11-06T07:14:00.554Z ef4175e9-fc8e-50f0-834a-2df7057875a7 Retrieved links content, 1 top-level links                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
| 2020-11-06 07:14:00.555 | [INFO] 2020-11-06T07:14:00.555Z ef4175e9-fc8e-50f0-834a-2df7057875a7 Subgraph BundleFQID(uuid='42d4f8d4-5422-4b78-adae-e7c3c2ef511c', version='2020-10-07T11:11:17.095956Z') is self-contained                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |
| 2020-11-06 07:14:00.555 | [INFO] 2020-11-06T07:14:00.555Z ef4175e9-fc8e-50f0-834a-2df7057875a7 Stitched together 1 subgraphs: {BundleFQID(uuid='42d4f8d4-5422-4b78-adae-e7c3c2ef511c', version='2020-10-07T11:11:17.095956Z')}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
| 2020-11-06 07:14:00.555 | [INFO] 2020-11-06T07:14:00.555Z ef4175e9-fc8e-50f0-834a-2df7057875a7 Retrieving 1 entities of type 'project' ...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
| 2020-11-06 07:13:58.188 | [INFO] 2020-11-06T07:13:58.188Z ef4175e9-fc8e-50f0-834a-2df7057875a7 Worker handling message {'action': 'add', 'notification': {'query': {}, 'subscription_id': 'cafebabe-feed-4bad-dead-beaf8badf00d', 'transaction_id': 'bc171f20-b71c-4f31-8fc9-de41e88424ab', 'match': {'bundle_uuid': '42d4f8d4-5422-4b78-adae-e7c3c2ef511c', 'bundle_version': '2020-10-07T11:11:17.095956Z'}}, 'catalog': 'dcp2'}, attempt #1 (approx).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |
| 2020-11-06 07:13:58.184 | START RequestId: ef4175e9-fc8e-50f0-834a-2df7057875a7 Version: $LATEST                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

**Comment by danielsotirhos** [comment]
All of these UUID (except `14c7a4c3`) are project UUID, not bundle UUID.

These errors appear to be a result of the first import/snapshot of the Contributor-generated matrices files which were staged with incorrect links.json filenames, using `{project_id}_{timestamp}_{project_id}.json` instead of `{bundle_id}_{timestamp}_{project_id}.json` (See this [staging area here](https://console.cloud.google.com/storage/browser/broad-dsp-monster-hca-prod-ucsc-storage/ucm/links)). The bundle version `2020-10-07T11:11:17.095956Z` from the error logs confirms this hypothesis.

This issue with the links.json filename was fixed for the files staged in the [current staging area](https://console.cloud.google.com/storage/browser/broad-dsp-monster-hca-prod-ucsc-storage/cgm_dcp2ebi).

Of the 16 bad UUID from this ticket, 15 match the 15 project id in the current staging area, and 1 (`14c7a4c3`) matches the link id of project `a815c84b` which is also 1 of the 15 projects.

```
c41dffbf-ad83-447c-a0e1-13e689d9b258
ad8902a0-8579-50a9-afce-2bb85610d414_2020-10-26T09:37:17.517006Z_c41dffbf-ad83-447c-a0e1-13e689d9b258.json

ad98d3cd-26fb-4ee3-99c9-8a2ab085e737
c4a7112c-b61c-510a-9871-0651f51aefaa_2020-10-26T09:37:17.517006Z_ad98d3cd-26fb-4ee3-99c9-8a2ab085e737.json
 
83f5188e-3bf7-4956-9544-cea4f8997756
9e434cfd-2263-5a23-a49a-d85a270d7040_2020-10-26T09:37:17.517006Z_83f5188e-3bf7-4956-9544-cea4f8997756.json
 
7027adc6-c9c9-46f3-84ee-9badc3a4f53b
a5d71125-fee9-5056-bfc8-3e93c82a8f32_2020-10-26T09:37:17.517006Z_7027adc6-c9c9-46f3-84ee-9badc3a4f53b.json
 
14c7a4c3-b01f-5923-afa5-ad2ce304ee6d *
14c7a4c3-b01f-5923-afa5-ad2ce304ee6d_2020-10-26T09:37:17.517006Z_a815c84b-8999-433f-958e-422c0720e00d.json
 
2086eb05-10b9-432b-b7f0-169ccc49d270
94565940-7b2f-5fd7-beb1-dc309687fe09_2020-10-26T09:37:17.517006Z_2086eb05-10b9-432b-b7f0-169ccc49d270.json
 
455b46e6-d8ea-4611-861e-de720a562ada
c480697b-d5ae-5830-9a94-a444a210ec69_2020-10-26T09:37:17.517006Z_455b46e6-d8ea-4611-861e-de720a562ada.json
 
95f07e6e-6a73-4e1b-a880-c83996b3aa5c
21e70ffa-b57c-5087-93d3-c49b9a4fc6d6_2020-10-26T09:37:17.517006Z_95f07e6e-6a73-4e1b-a880-c83996b3aa5c.json
 
a815c84b-8999-433f-958e-422c0720e00d
14c7a4c3-b01f-5923-afa5-ad2ce304ee6d_2020-10-26T09:37:17.517006Z_a815c84b-8999-433f-958e-422c0720e00d.json
 
b4a7d12f-6c2f-40a3-9e35-9756997857e3
3d7964fc-d0f9-548b-8a51-77ab19c33d48_2020-10-26T09:37:17.517006Z_b4a7d12f-6c2f-40a3-9e35-9756997857e3.json
 
c1a9a93d-d9de-4e65-9619-a9cec1052eaa
c69a3e74-916d-56b5-95cd-dcd6198244e7_2020-10-26T09:37:17.517006Z_c1a9a93d-d9de-4e65-9619-a9cec1052eaa.json
 
2ef3655a-973d-4d69-9b41-21fa4041eed7
4f2fc365-9f97-51ca-bbfe-fe30cefc333d_2020-10-26T09:37:17.517006Z_2ef3655a-973d-4d69-9b41-21fa4041eed7.json
 
f2fe82f0-4454-4d84-b416-a885f3121e59
a49ac587-4f29-5076-a130-ba0aeafd6881_2020-10-26T09:37:17.517006Z_f2fe82f0-4454-4d84-b416-a885f3121e59.json
 
38449aea-70b5-40db-84b3-1e08f32efe34
d0b2e31f-ab59-56eb-a972-b14bd422a423_2020-10-26T09:37:17.517006Z_38449aea-70b5-40db-84b3-1e08f32efe34.json
 
42d4f8d4-5422-4b78-adae-e7c3c2ef511c
57065113-5dfc-5e1a-9804-84808ad36b03_2020-10-26T09:37:17.517006Z_42d4f8d4-5422-4b78-adae-e7c3c2ef511c.json
 
f48e7c39-cc67-4055-9d79-bc437892840c
982ed605-2685-596c-9688-18c7d7e8a3ba_2020-10-26T09:37:17.517006Z_f48e7c39-cc67-4055-9d79-bc437892840c.json
 ```

**Comment by hannes-ucsc** [comment]
What are the next steps? Triage again? Provide estimate for resolution? Make this a stub and contact BI?

**Comment by theathorn** [comment]
Daniel identified 42 invalid links.json files which Raaid will soft delete and re-snapshot ([Slack thread](https://humancellatlas.slack.com/archives/C01360XN04S/p1604954969364400)).

**Comment by danielsotirhos** [comment]
Raaid has removed the 42 invalid rows from the links table and created a new snapshot `hca_dev_20200819___20201109_optimus`

**Comment by hannes-ucsc** [comment]
Snapshot `hca_dev_20200819___20201105_optimusb` has the same 16 errors, but only those errors. The remaining 7 errors (23 - 16) errors in https://github.com/DataBiosphere/azul/issues/2434 have disappeared so the fix for #2435 (PR #2438) seems to be effective.

**Comment by danielsotirhos** [comment]
Confirmed these 16 UUID are not present in the `hca_dev_20200819___20201109_optimus` snapshot.

```
SELECT * FROM hca_dev_20200819___20201109_optimus.links
WHERE links_id = 'c41dffbf-ad83-447c-a0e1-13e689d9b258'
OR links_id = 'ad98d3cd-26fb-4ee3-99c9-8a2ab085e737'
OR links_id = '83f5188e-3bf7-4956-9544-cea4f8997756'
OR links_id = '7027adc6-c9c9-46f3-84ee-9badc3a4f53b'
OR links_id = '14c7a4c3-b01f-5923-afa5-ad2ce304ee6d'
OR links_id = '2086eb05-10b9-432b-b7f0-169ccc49d270'
OR links_id = '455b46e6-d8ea-4611-861e-de720a562ada'
OR links_id = '95f07e6e-6a73-4e1b-a880-c83996b3aa5c'
OR links_id = 'a815c84b-8999-433f-958e-422c0720e00d'
OR links_id = 'b4a7d12f-6c2f-40a3-9e35-9756997857e3'
OR links_id = 'c1a9a93d-d9de-4e65-9619-a9cec1052eaa'
OR links_id = '2ef3655a-973d-4d69-9b41-21fa4041eed7'
OR links_id = 'f2fe82f0-4454-4d84-b416-a885f3121e59'
OR links_id = '38449aea-70b5-40db-84b3-1e08f32efe34'
OR links_id = '42d4f8d4-5422-4b78-adae-e7c3c2ef511c'
OR links_id = 'f48e7c39-cc67-4055-9d79-bc437892840c'
ORDER BY links_id
```

```
This query returned no results.
```
